packages/components/pagination/src/page-size-selector/page-size-selector.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import invariant from 'tiny-invariant';
+import uniqueId from 'lodash/uniqueId';
+import SelectInput from '@commercetools-uikit/select-input';
+import Spacings from '@commercetools-uikit/spacings';
+import Constraints from '@commercetools-uikit/constraints';
+import Label from '@commercetools-uikit/label';
+import messages from './messages';
+
+const mapRangeToListOfOptions = (pageSizeRange) => {
+  switch (pageSizeRange) {
+    case 's':
+      return [20, 50];
+    case 'm':
+      return [20, 50, 100];
+    case 'l':
+      return [200, 500];
+    default:
+      throw new Error(
+        `Invalid page size range "${pageSizeRange}", expected one of "s,m,l".`
+      );
+  }
+};
+
+const PageSizeSelector = (props) => {
+  const [pageSizeSelectorId] = React.useState(uniqueId('page-size-selector-'));
+
+  const options = mapRangeToListOfOptions(props.pageSizeRange);
+  const hasValidPageSizeOptions = options.includes(props.pageSize);
+
+  invariant(
+    hasValidPageSizeOptions,
+    `@commercetools-uikit/pagination: invalid page size ${
+      props.pageSize
+    }. It must be one of the values of the selected range in "${options.toString()}".`
+  );
+
+  const { onPageSizeChange } = props;
+  const handleSelectPerPage = React.useCallback(
+    (event) => {
+      onPageSizeChange(Number(event.target.value));
+    },
+    [onPageSizeChange]
+  );
+
+  return (
+    <Spacings.Inline alignItems="center">
+      <Constraints.Horizontal max={2}>
+        <SelectInput
+          id={pageSizeSelectorId}
+          name="page-size-selector"
+          value={props.pageSize.toString()}
+          options={options.map((option) => ({
```

Comment:
Nit: in some other parts we use a lot of `useCallback` etc. If we want to optimise for that we should `useMemo` here I think.